### PR TITLE
Added barcode_validator.py

### PR DESCRIPTION
**Pull request by Abinashbunty**
Added Google Cirq references

### Describe your change:
Added references to Google Cirq which is one of the most popular frameworks for developing quantum algorithms.

* [x] Add an algorithm?
* [ ] Fix a bug or typo in an existing algorithm?
* [x] Documentation change?

### Checklist:
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [x] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [x] All new Python files are placed inside an existing directory.
* [x] All filenames are in all lowercase characters with no spaces or dashes.
* [x] All functions and variable names follow Python naming conventions.
* [x] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [x] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [x] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.
* [ ] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

File: strings/barcode_validator.py [hunk 1]
```diff
@@ -0,0 +1,39 @@
+def eval_key(code: int):
```
[the remark this answers is not in the document]
Comment (Abinashbunty) [reply strings/barcode_validator.py]: Done

[the remark this answers is not in the document]
Comment (Abinashbunty) [reply strings/barcode_validator.py]: Done

[the remark this answers is not in the document]
Comment (Abinashbunty) [reply strings/barcode_validator.py]: Done

File: strings/barcode_validator.py [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+    while code != 0:
+        mult = 1 if checker else 3
+        s += mult * (code % 10)
+        code //= 10
+        checker = not checker
+
+    return (10 - (s % 10)) % 10
+
+
+def is_valid(code: int):
```
[the remark this answers is not in the document]
Comment (Abinashbunty) [reply strings/barcode_validator.py]: Done

[the remark this answers is not in the document]
Comment (Abinashbunty) [reply strings/barcode_validator.py]: Done

**Review by Abinashbunty — COMMENTED** [review]
All the necessary changes updated. Requesting review for merge.

**Comment by Abinashbunty** [comment]
@cclauss Requesting a review. 😄 Thanks!

File: strings/barcode_validator.py [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+if __name__ == "__main__":
+    import doctest
+
+    doctest.testmod()
+    """
+    Enter a barcode.
+    Displays whether the entered barcode is valid or invalid.
+    """
+    barcode = input("Enter a barcode: ")
+    number_barcode = int(barcode)
```
Comment (cclauss) [inline strings/barcode_validator.py:50]: As discussed in `CONTRIBUTING.md`…
```suggestion
    barcode = int(input("Enter a barcode: ").strip())
```

File: strings/barcode_validator.py [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+    while code != 0:
+        mult = 1 if checker else 3
+        s += mult * (code % 10)
+        code //= 10
+        checker = not checker
+
+    return (10 - (s % 10)) % 10
+
+
+def is_valid(code: int) -> bool:
```
Comment (cclauss) [inline strings/barcode_validator.py:25]:
```suggestion
def is_barcode_valid(barcode: int) -> bool:
```

**Review by cclauss — CHANGES_REQUESTED** [review]

File: strings/barcode_validator.py [hunk 5]
```diff
@@ -0,0 +1,54 @@
+def eval_key(code: int) -> int:
+    """
+    Returns the last digit of barcode by excluding the last digit first
+    and then computing to reach the actual last digit from the remaining
+    12 digits.
+
+    >>> eval_key(8718452538119)
```
[the remark this answers is not in the document]
Comment (cclauss) [reply strings/barcode_validator.py:7]: Also, "abcdefghijklm"

File: strings/barcode_validator.py [hunk 6]
```diff
@@ -0,0 +1,54 @@
+def eval_key(code: int) -> int:
```
[the remark this answers is not in the document]
Comment (Abinashbunty) [reply strings/barcode_validator.py]: Done

File: strings/barcode_validator.py (hunk 5 above)
[the remark this answers is not in the document]
Comment (Abinashbunty) [reply strings/barcode_validator.py]: Done

File: strings/barcode_validator.py (hunk 3 above)
Comment (Abinashbunty) [reply strings/barcode_validator.py]: Done

File: strings/barcode_validator.py [hunk 7]
```diff
@@ -0,0 +1,86 @@
...
+    >>> get_barcode("8718452538119")
+    8718452538119
+
+    >>> get_barcode("dwefgiweuf")
+    Traceback (most recent call last):
+    ...
+    ValueError: Barcode 'dwefgiweuf' has alphabets in it
+    """
+    if str(barcode).isalpha():
+        raise ValueError("Barcode '{}' has alphabets in it".format(barcode))
```
[the remark this answers is not in the document]
Comment (Abinashbunty) [reply strings/barcode_validator.py]: Done

File: strings/barcode_validator.py (hunk 4 above)
Comment (Abinashbunty) [reply strings/barcode_validator.py]: Done

**Comment by Abinashbunty** [comment]
@cclauss Done with the changes requested. 😄

**Review by cclauss — APPROVED** [review]

**Comment by Abinashbunty** [comment]
Thanks @cclauss 😊 Will it be merged automatically?

**Pull request merged**